src/delay.rs

```diff
@@ -31,11 +33,17 @@ impl Delay {
         let ticks_per_us = freq / 1_000_000_u32;
         Delay { ticks_per_us, syst }
     }
+
+    /// Wait for the given time.
+    ///
+    /// Note that durations above `u32::MAX` microseconds will be clamped at `u32::MAX`.
     pub fn delay<T>(&mut self, delay: T)
     where
-        T: Into<MicroSeconds>,
+        T: TryInto<Microseconds>,
```

Comment:
This is so we can use e.g. `1.second()` or `300.milliseconds()` in calling code. We could also use this pattern in other hertz-accepting places. There's at least one demo that replaces `1.khz()` with `100_000.Hz()` which I don't like.